src/com/amazon/corretto/crypto/provider/SecretKeyGenerator.java

```diff
@@ -73,13 +73,7 @@ private DefaultSecureRandomSupplier() {
 
     @Override
     public SecureRandom get() {
-      try {
-        return SecureRandom.getInstance(
-            LibCryptoRng.ALGORITHM_NAME, AmazonCorrettoCryptoProvider.INSTANCE);
-      } catch (final NoSuchAlgorithmException e) {
-        throw new AssertionError(
-            LibCryptoRng.ALGORITHM_NAME + " is not a provided by AmazonCorrettoCryptoProvider", e);
-      }
+      return new LibCryptoRng();
```

Comment:
Just want to make sure, if we are NOT in FIPS mode then we still want to use `LibCryptoRng()`? Are there any cases where we'd want SecretKeyGenerator to fall back to non-ACCP vended SecureRandom?

Reply:
> if we are NOT in FIPS mode then we still want to use LibCryptoRng()?

I would argue yes. The reason is that SecretKey material should be derived from as high-quality of a source of randomness as possible. This also fits the pattern we establish in other asymmetric key (pair) generators of always using AWS-LC's DRBG (though in those cases it happens internally within AWS-LC).

> Are there any cases where we'd want SecretKeyGenerator to fall back to non-ACCP vended SecureRandom?

Perhaps in a scenario similar to the one that motivated PR #286 (latency-sensitive key generation with high rate of thread creation). However, @amirhosv points out that he wrote SecretKeyGenerator in such a way that the SecureRandom implementation used is configurable. This change just affects the default implementation used.

Reply:
It's part of JCE/JCA. [KeyGenerator::init](https://docs.oracle.com/javase/8/docs/api/javax/crypto/KeyGenerator.html#init-java.security.spec.AlgorithmParameterSpec-java.security.SecureRandom-) can accept `SecureRandom`. In case no SecureRandom is provided when [KeyGenerator::generateKey](https://docs.oracle.com/javase/8/docs/api/javax/crypto/KeyGenerator.html#generateKey--) is invoked, we fall back on `LibCryptoRng`.